P202.Training.Test/RoleServiceTest.cs

```diff
@@ -0,0 +1,139 @@
+﻿using AutoMapper;
+using Moq;
+using P202.Training.Data.Entities;
+using P202.Training.Data.Repositories;
+using P202.Training.Domain;
+using System.Collections.Generic;
+using System.Linq;
+using Xunit;
+
+namespace P202.Training.Test
+{
+    public class RoleServiceTest
+    {
+        [Fact]
+        public void CreateRole_PersistsChange()
+        {
+            // Arrange
+            var roleRepositoryMock = new Mock<IRoleRepository>();
+            var mappingServiceMock = new Mock<IMapper>();
+
+            var newRole = new Role
+            {
+                Id = 1,
+                Name = "Admin"
+            };
+
+            roleRepositoryMock.Setup(r => r.CreateRole(It.IsAny<Role>()));
+            mappingServiceMock.Setup(m => m.Map<Role, Role>(It.IsAny<Role>()))
+                            .Returns(newRole);
+            var mappingServiceMockObject = mappingServiceMock.Object;
+
+            // Act
+            var roleRepositoryMockObject = roleRepositoryMock.Object;
+            roleRepositoryMockObject.CreateRole(newRole);
```

Comment:
I'm almost sure this line is not needed. You shouldn't be calling mock methods from the UT. Just performing setups and verifications

Reply:
Done!